app/charts/map/map.tsx

```diff
@@ -117,37 +128,87 @@ export const MapComponent = () => {
     }
   }, []);
 
-  const zoomToShapes = useCallback(() => {
-    if (features.areaLayer?.shapes) {
-      const bbox = geoBounds(features.areaLayer.shapes);
+  const [isMapLoaded, setIsMapLoaded] = useState(false);
+  const hasSetInitialZoom = useRef<boolean>();
 
-      setViewState(constrainZoom(viewState, bbox, { padding: 48 }));
-    }
-  }, [viewState, features.areaLayer?.shapes]);
+  const getBBox = useCallback(
+    (shapes?: AreaLayer["shapes"], symbols?: SymbolLayer["points"]) => {
+      let shapesBbox: BBox | undefined;
+      let symbolsBbox: BBox | undefined;
 
-  const zoomToSymbols = useCallback(() => {
-    if (features.symbolLayer?.points) {
-      const visiblePoints = features.symbolLayer.points.filter(
-        (d) => d.properties.observation !== undefined
-      );
-      const [minLng, maxLng] = extent(visiblePoints, (d) => d.coordinates[0]);
-      const [minLat, maxLat] = extent(visiblePoints, (d) => d.coordinates[1]);
-      const bbox = [
-        [minLng, minLat],
-        [maxLng, maxLat],
-      ] as [[number, number], [number, number]];
+      if (shapes) {
+        shapesBbox = geoBounds(shapes);
+      }
+
+      if (symbols) {
+        const visiblePoints = symbols.filter(
+          (d) => d.properties.observation !== undefined
+        );
+        const [minLng, maxLng] = extent(visiblePoints, (d) => d.coordinates[0]);
+        const [minLat, maxLat] = extent(visiblePoints, (d) => d.coordinates[1]);
+
+        symbolsBbox = [
+          [minLng, minLat],
+          [maxLng, maxLat],
+        ] as BBox;
+      }
+
+      if (shapesBbox !== undefined) {
+        if (symbolsBbox !== undefined) {
+          const [minLng, maxLng] = [
+            Math.min(shapesBbox[0][0], symbolsBbox[0][0]),
+            Math.max(shapesBbox[1][0], symbolsBbox[1][0]),
+          ];
+          const [minLat, maxLat] = [
+            Math.min(shapesBbox[0][1], symbolsBbox[0][1]),
+            Math.max(shapesBbox[1][1], symbolsBbox[1][1]),
+          ];
+          const bbox = [
+            [minLng, minLat],
+            [maxLng, maxLat],
+          ] as BBox;
+
+          return bbox;
+        } else {
+          return shapesBbox;
+        }
+      } else {
+        return symbolsBbox;
+      }
+    },
+    []
+  );
```

Comment:
Can be extracted out of the component if there are no dependencies.

Reply:
Right, will do!